rfcs/0013-avalanche.md

```diff
@@ -0,0 +1,98 @@
+
+
+## New Design
+
+                                     .-------------.
+                                     |             |
+                       .-------------+   Leader    +══════════════╗
+                       |             |             |              ║
+                       |             `-------------`              ║
+                       v                                          v
+                .-------------.                            .-------------.
+                |             +--------------------------->|             |
+           .----+ Validator 1 |                            | Validator 2 +═══╗
+           |    |             |<═══════════════════════════+             |   ║
+           |    `------+------`                            `------+------`   ║
+           |           |                                          ║          ║
+           |           `------------------------------.           ║          ║
+           |                                          |           ║          ║
+           |                     ╔════════════════════════════════╝          ║
+           |                     ║                    |                      ║
+           V                     v                    V                      v
+    .-------------.       .-------------.       .-------------.       .-------------.
+    |             |       |             |       |             |       |             |
+    | Validator 3 +------>| Validator 4 +══════>| Validator 5 +------>| Validator 6 |
+    |             |       |             |       |             |       |             |
+    `-------------`       `-------------`       `-------------`       `------+------`
+           ^                                                                 ║
+           ║                                                                 ║
+           ╚═════════════════════════════════════════════════════════════════╝
+
+#### What’s needed -
+The main goal of the new design is to enable Layer-2 nodes in the replication process. There are a few challenges with this and we should be able 
+to address some of them.  
+
+* Most importantly, there needs to be a mechanism to figure out which nodes belong in what layer of this avalanche. We need a Stake weighted selection for the nodes. This is missing and is very important. 
+* Leaders should only broadcast to layer 1 nodes and the network should be split (ClusterInfo can provide leader’s layer-1 tvu peers) 
+Layer 1 nodes should broadcast to all peers at the same layer and to a subset of peers at lower layers. 
+(ClusterInfo can provide Layer 1’s peers and layer 2 sections)
+* Layer 2 nodes should only retransmit amongst themselves. They should also be sectioned off and only need to communicate with each other instead of all layer 2 nodes. 
```

Comment:
* layer 2 nodes should be able to recover the full ledger if M erasure codes are used, and less than M layer 1 nodes failed.